lib/nexpose/asset.rb

```diff
@@ -263,6 +265,38 @@ def eql?(other)
     end
   end
 
+  # Unique system identifiers on an asset.
+  #
+  class UniqueIdentifier < APIObject
+    # The source name for the uniuqe identifier.
+    attr_reader :source
+    # Unique identifier of the user as determined by the asset (not Nexpose).
+    attr_reader :id
+
+    def initialize(source = nil, id = nil)
+      @id, @source = id, source
+    end
+
+    def to_h
+      { source: source,
+        id: id}
+    end
+
+    def <=>(other)
+      c = source <=> other.source
+      return c unless c == 0
+      c = id <=> other.id
```

Comment:
@hwilson-r7 for local barking you can use the `rubocop-git` gem 😄  (which I should probably setup a commit hook for)